compatible/src/main/java/io/seata/saga/engine/config/DbStateMachineConfig.java

```diff
@@ -16,6 +16,206 @@
+
+            if (StringUtils.hasLength(getSagaJsonParser())) {
+                ParamsSerializer paramsSerializer = new ParamsSerializer();
+                paramsSerializer.setJsonParserName(getSagaJsonParser());
+                dbStateLogStore.setParamsSerializer(paramsSerializer);
+            }
+
+            if (sagaTransactionalTemplate == null) {
+                DefaultSagaTransactionalTemplate defaultSagaTransactionalTemplate = new DefaultSagaTransactionalTemplate();
+                defaultSagaTransactionalTemplate.setApplicationContext(getApplicationContext());
+                defaultSagaTransactionalTemplate.setApplicationId(applicationId);
+                defaultSagaTransactionalTemplate.setTxServiceGroup(txServiceGroup);
+                defaultSagaTransactionalTemplate.setAccessKey(accessKey);
+                defaultSagaTransactionalTemplate.setSecretKey(secretKey);
+                defaultSagaTransactionalTemplate.afterPropertiesSet();
+                sagaTransactionalTemplate = defaultSagaTransactionalTemplate;
+            }
+
+            dbStateLogStore.setSagaTransactionalTemplate(sagaTransactionalTemplate);
+
+            setStateLogStore(StateLogStoreImpl.wrap(dbStateLogStore));
+        }
+
+        if (getStateLangStore() == null) {
+            DbStateLangStore dbStateLangStore = new DbStateLangStore();
+            dbStateLangStore.setDataSource(dataSource);
+            dbStateLangStore.setTablePrefix(tablePrefix);
+            dbStateLangStore.setDbType(dbType);
+
+            setStateLangStore(dbStateLangStore);
+        }
+
+        //must execute after StateLangStore initialized
+        super.afterPropertiesSet();
+    }
+
+    @Override
+    public void destroy() throws Exception {
+        if ((sagaTransactionalTemplate != null) && (sagaTransactionalTemplate instanceof DisposableBean)) {
+            ((DisposableBean) sagaTransactionalTemplate).destroy();
+        }
+    }
+
+    public DataSource getDataSource() {
+        return dataSource;
+    }
+
+    public void setDataSource(DataSource dataSource) {
+        this.dataSource = dataSource;
+    }
+
+    public String getApplicationId() {
+        return applicationId;
+    }
+
+    public void setApplicationId(String applicationId) {
+        this.applicationId = applicationId;
+    }
+
+    public String getTxServiceGroup() {
+        return txServiceGroup;
+    }
+
+    public void setTxServiceGroup(String txServiceGroup) {
+        this.txServiceGroup = txServiceGroup;
+    }
+
+    public String getAccessKey() {
+        return accessKey;
+    }
+
+    public void setAccessKey(String accessKey) {
+        this.accessKey = accessKey;
+    }
+
+    public String getSecretKey() {
+        return secretKey;
+    }
+
+    public void setSecretKey(String secretKey) {
+        this.secretKey = secretKey;
+    }
+
+    public void setSagaTransactionalTemplate(SagaTransactionalTemplate sagaTransactionalTemplate) {
```

Comment:
It looks like the parameter should be io.seata.saga.engine.tm.SagaTransactionalTemplate.

Reply:
compatibility 'sagaTransactionalTemplate' will be more complicated. For example, if you need to be compatible with sagaTransactionalTemplate ,must compatible TransactionalExecutor -> compatible with TransactionInfo -> compatible with TransactionInfo RollbackRule and LockStrategyMode etc class in TransactionInfo.

Considering that it is rarely used, it is decided not to be compatible in this pr.